.github/workflows/metrics.yml

```diff
@@ -36,12 +33,24 @@ jobs:
         if: matrix.backend == 'postgres'
         run: |
           sudo apt-get update
-          sudo apt-get install -y libpq-dev postgresql
-          echo "host    all             all             127.0.0.1/32            md5" > sudo tee -a /etc/postgresql/10/main/pg_hba.conf
-          sudo service postgresql restart && sleep 3
+          sudo DEBIAN_FRONTEND=noninteractive apt-get --purge remove postgresql\* -y
+          sudo apt-get install gnupg2 -y
+          curl -fsSL https://www.postgresql.org/media/keys/ACCC4CF8.asc|sudo gpg --dearmor -o /etc/apt/trusted.gpg.d/postgresql.gpg
+          sudo sh -c 'echo "deb https://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'
+          sudo apt-get update
+          sudo apt-get install -y libpq-dev postgresql-$PG_VERSION
+          sudo tee /etc/postgresql/$PG_VERSION/main/pg_hba.conf <<'EOF'
+          local   all             postgres                                peer
+          local   all             all                                     peer
+          host    all             all             127.0.0.1/32            trust
+          host    all             all             ::1/128                 trust
+          EOF
+          sudo service postgresql start $PG_VERSION && sleep 3
           sudo -u postgres psql -c "ALTER USER postgres PASSWORD 'postgres';"
-          sudo service postgresql restart && sleep 3
-          echo 'DATABASE_URL=postgres://postgres:postgres@localhost/' >> $GITHUB_ENV
+          sudo service postgresql restart $PG_VERSION && sleep 3
+          echo 'DATABASE_URL=postgres://postgres:postgres@localhost:5432/' >> $GITHUB_ENV
+        env:
```

Comment:
Can we move that to the top of the script block? I would expect that people (including me) will miss that place in the future otherwise.

Reply:
Done.